libaleth-interpreter/VM.cpp

```diff
@@ -380,17 +380,20 @@ void VM::interpretCases()
             m_runGas = m_rev >= EVMC_TANGERINE_WHISTLE ? 5000 : 0;
             evmc_address destination = toEvmC(asAddress(m_SP[0]));
 
-            // After EIP158 zero-value suicides do not have to pay account creation gas.
-            u256 const balance =
-                fromEvmC(m_context->host->get_balance(m_context, &m_message->destination));
-            if (balance > 0 || m_rev < EVMC_SPURIOUS_DRAGON)
+            // Self-destructs only have gas cost starting with Tangerine Whistle
+            if (m_rev >= EVMC_TANGERINE_WHISTLE)
             {
-                // After EIP150 hard fork charge additional cost of sending
-                // ethers to non-existing account.
-                int destinationExists =
-                    m_context->host->account_exists(m_context, &destination);
-                if (m_rev >= EVMC_TANGERINE_WHISTLE && !destinationExists)
-                    m_runGas += VMSchedule::callNewAccount;
+                // After EIP158 zero-value suicides do not have to pay account creation gas.
+                if (m_rev == EVMC_TANGERINE_WHISTLE ||
+                    fromEvmC(m_context->host->get_balance(m_context, &m_message->destination)) > 0)
+                {
+                    // After EIP150 hard fork charge additional cost of sending
+                    // ethers to non-existing account.
+                    int const destinationExists =
```

Comment:
This should be `bool` now, maybe it was `int` because of some older version of EVMC

Reply:
or could be just inside `if (...)` because now it has only this condition and so is simpler